Курсовая работа 1

VadyaTheFirst opened the pull request:
Базовый уровень

VadyaTheFirst committed:
Курсовая работа 1
Базовый уровень

VadyaTheFirst committed:
Курсовая работа 1
Базовый уровень
Добавил остальные 7 элементов

DmitryN10 commented on src/EmployeesAction.java:
```diff
@@ -0,0 +1,55 @@
+public class EmployeesAction {
+
+
+    public static void EmployeesCout(Employee[] arr) {
```
Стоит именовать все методы с маленькой буквы- camelCase

VadyaTheFirst replied:
Сделал

DmitryN10 commented on src/EmployeesAction.java:
```diff
@@ -0,0 +1,55 @@
+        var max=arr[0].getSalary();
+        byte k=0;
+        for (byte i = 1; i < arr.length; i++) {
+            {
```
23 и 29 строка лишняя

VadyaTheFirst replied:
Сделал

DmitryN10 commented on src/EmployeesAction.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    public  static void AvarageSalary(Employee[] arr){
+        var averageSalary=EmployeesAction.SumSalary(arr)/ arr.length;
```
А если массив не полностью заполнен? Тогда надо узнать количество сотрудников в массиве

VadyaTheFirst replied:
Дописал метод лля подсчета числа непустых элементов массива.
PS: лучше это добавить в требования

DmitryN10 commented on src/EmployeesAction.java:
```diff
@@ -0,0 +1,55 @@
+
+    public  static void AvarageSalary(Employee[] arr){
+        var averageSalary=EmployeesAction.SumSalary(arr)/ arr.length;
+        System.out.println("Средняя зарплата: "+ averageSalary);
```
во всех методах перед тем как вызывать arr[i].getSalary(); надо убедиться, что arr[i] != null

VadyaTheFirst replied:
А почему внутри метода getSalary проверка this.salary !=null не работает?

VadyaTheFirst replied:
Обсудили

VadyaTheFirst committed:
Курсовая работа 1
Базовый уровень
Добавил проверку на null-значения и число непустых элементов для среднего

VadyaTheFirst committed:
Курсовая работа 1
Базовый уровень
Добавил продвинутую сложность

DmitryN10 commented on src/EmployeesAction.java:
```diff
@@ -0,0 +1,212 @@
+    public static void nullCout(Employee[] arr) {
+        for (byte i = 0; i < arr.length; i++) {
+            if (arr[i] == null) {
+                throw new NullPointerException("Нет данных о сотруднике с идентификаторром " + i);
```
Это как-то странно пробрасывать ошибку.
Правильнее все-таки выводить в консоль подобную инфу

DmitryN10 commented on src/EmployeesAction.java:
```diff
@@ -0,0 +1,212 @@
+    }
+
+
+    public static void employeesCout(Employee[] arr) {
```
Название метода стоит изменить. Не информаттивно

DmitryN10 commented on src/EmployeesAction.java:
```diff
@@ -0,0 +1,212 @@
+        } else {
+
+            for (byte i = 0; i < arr.length; i++) {
+                if (arr[i].getDepartment() == number) {
```
если arr[i] будет null, тогда NPE

DmitryN10 replied:
Я здесь имею ввиду, если массив у тебя не заполнен, значит будут ячейки в массиве со значением null.
А знчит при вызове arr[i].getDepartment() будет NullPointerException, тк на arr[i]  =null

VadyaTheFirst committed:
Курсовая работа 1
Базовый уровень
